appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/EnvironmentProperty.java

```diff
@@ -386,50 +386,49 @@ public void print(StringBuilder toStringBuilder) {
     }
 
     private Object getObjectFromString(String string, Class type) {
-        if (type == null && !this.isBoundsChecking()) {
+        if (type == null && !Descriptor.isBoundsChecking()) {
             Object obj = getValueObjectUsingAllowedTypes(string);
             if (obj != null) return obj;
         }
-	if (string == null || ("".equals(string) && !type.equals(String.class))) {
-	    return null;
-	}
-	try {
-            if (String.class.equals(type)) {
-		return string;
-            } else if (Boolean.class.equals(type)) {
-		return Boolean.valueOf(string);
-	    } else if (Integer.class.equals(type)) {
-		return Integer.valueOf(string);
-	    } else if (Double.class.equals(type)) {
-		return new Double(string);
-	    } else if (Float.class.equals(type)) {
-		return new Float(string);
-	    } else if (Short.class.equals(type)) {
-		return Short.valueOf(string);
-	    } else if (Byte.class.equals(type)) {
-		return Byte.valueOf(string);
-	    } else if (Long.class.equals(type)) {
-		return Long.valueOf(string);
-	    } else if (Character.class.equals(type)) {
+        if (string == null || ("".equals(string) && type != String.class)) {
```

Comment:
Using the proper way of comparing `Class`es this also removes the possible NPE for `type` and maps any `null` input for `type` to `null` output which seems reasonable. 

Below I corrected funny indentation, comparison and unnecessary `else` nesting.